plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -232,6 +232,11 @@ public IcebergMetadata(
     @Override
     public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
+        // We rely on transaction level metadata cache to avoid incurring more communication to metastore
```

Comment:
This seems wrong, as there is no transaction level metastore cache for Iceberg

Reply:
addressed in https://github.com/trinodb/trino/pull/8165

Reply:
from offline conversation:

> We are now calling metastore.getTable() twice, which can be a big performance hit. It needs to be fixed to only call it once.